discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py

```diff
@@ -52,6 +52,22 @@ def _create_tf_folder(self, name: str, platform: str):
 
         return os.path.join(tf_folder, consts.Platforms.BARE_METAL)
 
+    def _get_disk_encryption_appliance(self):
+        assert (
+            self._entity_config.disk_encryption_mode == consts.DiskEncryptionMode.TPM_VERSION_2
```

Comment:
It's only exists on `BaseClusterConfig`.
This class can also get `BaseInfraEnvConfig`.
To my opinion you can:

1. Add it to `BaseInfraEnvConfig` (e.g. `BaseEntityConfig`) if it's also related to InfraEnv
2. Verify it has this property first in this function (not in `_get_specific_tf_entity_params`)
3. Pass `BaseInfraEnvConfig` to the function instead of using it via self

Reply:
I'll make it agnostic to the config it uses
thanks!